client/state/reader/follows/reducer.js

```diff
@@ -64,28 +66,48 @@ function updatePostSubscription( state, { payload, type } ) {
 	};
 }
 
-/**
- * Tracks all known list objects, indexed by list ID.
- *
- * @param  {Array} state  Current state
- * @param  {Object} action Action payload
- * @return {Array}        Updated state
- */
 export const items = createReducer( {}, {
-	[ READER_FOLLOW ]: ( state, action ) => {
+	[ READER_RECORD_FOLLOW ]: ( state, action ) => {
 		const urlKey = prepareComparableUrl( action.payload.url );
 		return {
 			...state,
 			[ urlKey ]: merge( {}, state[ urlKey ], { is_following: true } ),
 		};
 	},
-	[ READER_UNFOLLOW ]: ( state, action ) => {
+	[ READER_RECORD_UNFOLLOW ]: ( state, action ) => {
 		const urlKey = prepareComparableUrl( action.payload.url );
 		return {
 			...state,
 			[ urlKey ]: merge( {}, state[ urlKey ], { is_following: false } ),
 		};
 	},
+	[ READER_FOLLOW ]: ( state, action ) => {
+		const urlKey = prepareComparableUrl( action.payload.feedUrl );
+		return {
+			...state,
+			[ urlKey ]: merge(
+				{ feed_URL: action.payload.feedUrl },
```

Comment:
this order doesn't seem correct.
Why would we want to potentially override the  feed_URL from the payload with the one currently in state?